### PR TITLE
Feature/error detail implements i error detail

### PR DESCRIPTION
**Pull request by willie84**
Kindly review the implementation  of IErrorDetail so that we can do a final release.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -34,16 +34,16 @@
...
     <jersey-version>2.22.1</jersey-version>
     <hibernate-validator-version>5.4.3.Final</hibernate-validator-version>
     <el-version>2.2.4</el-version>
     <swagger-maven-plugin-version>3.1.3</swagger-maven-plugin-version>
     <logback-version>1.0.13</logback-version>
     <major-version>5</major-version>
-    <minor-version>25</minor-version>
-    <patch-version>0</patch-version>
+    <minor-version>26</minor-version>
+    <patch-version>0-RC3</patch-version>
```
Comment (Shaaheen) [inline pom.xml:46]: final

File: pom.xml [hunk 2]
```diff
@@ -34,16 +34,16 @@
 
   <properties>
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-    <service-interface-base-version>3.28.0</service-interface-base-version>
+    <service-interface-base-version>3.39.0-RC4</service-interface-base-version>
```
Comment (Shaaheen) [inline pom.xml:37]: final

File: src/main/java/io/electrum/airtime/api/model/ErrorDetail.java [hunk 3]
```diff
@@ -142,6 +144,21 @@ public void setOriginalId(String originalId) {
       this.originalId = originalId;
    }
 
+
+   /**
+    * Implemented from IErrorDetail interfaces.
+    **/
+   @Override
+   public void setErrorTypeFromString(String s) throws Exception {
+      this.errorType = ErrorType.valueOf(s);
```
Comment (Shaaheen) [inline src/main/java/io/electrum/airtime/api/model/ErrorDetail.java:153]: name variable a bit better, rather than `s` - probably name `errorTypeAsString`

**Review by Shaaheen — CHANGES_REQUESTED** [review]
- Needs some tests. Can probably add in the https://github.com/electrumpayments/airtime-service-interface/blob/master/src/test/java/io/electrum/airtime/api/ModelsTest.java
- Please update release notes with what changes you made including the service interface bump that you did as that was quite a hefty jump (10 minor versions)

File: pom.xml (hunk 1 above)
Comment (VeeneshJeena) [reply pom.xml:46]: remember to finalise

File: release-notes.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    - New Features: Changes that would, in the absence of any breaking changes, constitute a minor version release.
+    - Fixed: Bugfixes that would, in the absence of any new features or breaking changes, constitute a patch version release.
+    - Deprecated: Any classes or methods that have been deprecated.
+  * Make use of Markdown formatting:
+    - Run ‘$curl cheat.sh/markdown’ from your command line to get a quick overview of markdown.
+    - Use the convention of enclosing class, variable and method names in backticks so that they render as monospace.
+    - Try and avoid special characters as far as possible
+-->
+
+## Version 5.26.0-RC3 - 03 June 2022
```
Comment (VeeneshJeena) [inline release-notes.md:15]: finalise the version

File: src/main/java/io/electrum/airtime/api/model/ErrorDetail.java [hunk 5]
```diff
@@ -142,6 +144,21 @@ public void setOriginalId(String originalId) {
       this.originalId = originalId;
    }
 
+
+   /**
+    * Implemented from IErrorDetail interfaces.
+    **/
+   @Override
+   public void setErrorTypeFromString(String errorTypeFromString) throws Exception {
+      this.errorType = ErrorType.valueOf(errorTypeFromString);
```
Comment (VeeneshJeena) [inline src/main/java/io/electrum/airtime/api/model/ErrorDetail.java:153]:
```suggestion
   public void setErrorTypeFromString(String errorTypeAsString) throws Exception {
      this.errorType = ErrorType.valueOf(errorTypeAsString);
```

File: src/main/java/io/electrum/airtime/api/model/ErrorDetail.java [hunk 6]
```diff
@@ -142,6 +144,21 @@ public void setOriginalId(String originalId) {
       this.originalId = originalId;
    }
 
+
+   /**
+    * Implemented from IErrorDetail interfaces.
```
Comment (VeeneshJeena) [inline src/main/java/io/electrum/airtime/api/model/ErrorDetail.java:149]: This Javadoc is incorrect. This should document what this method does.

**Review by VeeneshJeena — CHANGES_REQUESTED** [review]
small changes required

File: release-notes.md [hunk 7]
```diff
@@ -12,11 +12,9 @@
     - Try and avoid special characters as far as possible
 -->
 
-## Version 5.26.0-RC3 - 03 June 2022
+## Version 5.26.1 - 03 June 2022
```
Comment (VeeneshJeena) [inline release-notes.md:15]: version number is incorrect, It should be 5.26.0.
Update the date to today's date.

File: src/main/java/io/electrum/airtime/api/model/ErrorDetail.java [hunk 8]
```diff
@@ -146,11 +146,11 @@ public void setOriginalId(String originalId) {
 
 
    /**
-    * Implemented from IErrorDetail interfaces.
+    * Implemented from IErrorDetail where the errorType is passed string and the value is evaluated from the Enum.
```
Comment (VeeneshJeena) [inline src/main/java/io/electrum/airtime/api/model/ErrorDetail.java:149]:
```suggestion
    * Converts an errorType from a String representation to the enum equivalent.
```

**Review by VeeneshJeena — CHANGES_REQUESTED** [review]
some small comments

**Comment by willie84** [comment]
v5.26.0 [release-type final]

**Pull request merged**